botocore/credentials.py

```diff
@@ -1919,10 +1920,21 @@ def _retrieve_or_fail(self):
         )
 
     def _build_headers(self):
-        auth_token = self._environ.get(self.ENV_VAR_AUTH_TOKEN)
-        if auth_token is not None:
+        auth_token = None
+        if self.ENV_VAR_AUTH_TOKEN_FILE in self._environ:
+            auth_token_file_path = self._environ[self.ENV_VAR_AUTH_TOKEN_FILE]
+            with open(auth_token_file_path) as token_file:
+                auth_token = token_file.read()
+        elif self.ENV_VAR_AUTH_TOKEN in self._environ:
+            auth_token = self._environ[self.ENV_VAR_AUTH_TOKEN]
+        if auth_token:
```

Comment:
This got more specific, what was the reason for changing it from `is not None`?

Reply:
If the string winds up being empty - say due to an empty file being read.  This should probably be changed back to is not none and then we should validate it is a non empty string in the validate function instead.  Should we also be adding debug statements to let users know if the token came from a file or environment variable?